source/extensions/filters/http/ext_proc/config.cc

```diff
@@ -9,23 +9,24 @@ namespace Extensions {
 namespace HttpFilters {
 namespace ExternalProcessing {
 
-Http::FilterFactoryCb ExternalProcessingFilterConfig::createFilterFactoryFromProtoTyped(
+absl::StatusOr<Http::FilterFactoryCb>
+ExternalProcessingFilterConfig::createFilterFactoryFromProtoTyped(
     const envoy::extensions::filters::http::ext_proc::v3::ExternalProcessor& proto_config,
-    const std::string& stats_prefix, Server::Configuration::FactoryContext& context) {
+    const std::string& stats_prefix, DualInfo dual_info,
+    Server::Configuration::ServerFactoryContext& context) {
   const uint32_t message_timeout_ms =
       PROTOBUF_GET_MS_OR_DEFAULT(proto_config, message_timeout, DefaultMessageTimeoutMs);
   const uint32_t max_message_timeout_ms =
       PROTOBUF_GET_MS_OR_DEFAULT(proto_config, max_message_timeout, DefaultMaxMessageTimeoutMs);
   const auto filter_config = std::make_shared<FilterConfig>(
       proto_config, std::chrono::milliseconds(message_timeout_ms), max_message_timeout_ms,
-      context.scope(), stats_prefix,
-      Envoy::Extensions::Filters::Common::Expr::getBuilder(context.serverFactoryContext()),
-      context.serverFactoryContext());
+      dual_info.scope, stats_prefix, dual_info.is_upstream,
+      Envoy::Extensions::Filters::Common::Expr::getBuilder(context), context);
 
-  return [filter_config, grpc_service = proto_config.grpc_service(),
-          &context](Http::FilterChainFactoryCallbacks& callbacks) {
+  return [filter_config, grpc_service = proto_config.grpc_service(), &context,
+          dual_info](Http::FilterChainFactoryCallbacks& callbacks) {
```

Comment:
I guess it needs to be `pass-by-copy` because the object is created as temporary object `DualInfo(context)` at the very top of the caller site?

This object is cheap to copy as it only has ref member and a bool

Reply:
Right, the lifetime model of dual_info make it has to be passed by copy.

Reply:
Thanks for confirming!